andela/activo-api
# #158757927 Create pre-commit hook

#### What does this PR do?
- Migrate the project from managing dependencies in a `requirements.txt` file to using `pipfiles`. This separates development and production dependencies in an elegant and easy to manage way.
- Create a pre-commit hook to auto format python code to `pep8` using `yapf` then check for residual linting errors using `pylint`.

#### Description of Task to be completed?
- Create way to easily manage development and production dependencies. The linting and formatting dependencies should reside in the development dependencies separate from production dependencies.

- Create pre commit hook that automatically formats the code to `pep8` and highlights linting errors not automatically resolved.

#### How should this be manually tested?
- Create a local copy with the following commands: `git fetch origin ch-create-pre-commit-hook-158757927:ch-create-pre-commit-hook-158757927`, then `git checkout ch-create-pre-commit-hook-158757927`
- Follow the instructions on how to setup the pre-commit hook into your own repository. These instructions are in the `Contribution Guide` section in the Readme file.
- Once your pre-commit hook has been setup, create a test branch and make changes contrary to `pep8` to any `.py` file and attempt to commit it without staging it. You should get an error stating `You have unstaged changes to some files in your commit;`
- Add the changed files to staging and attempt to make a commit. The pre-commit hook should be called and auto-format the `.py` files to `pep8` then call `pylint` on the files. If there are any linting issues, the commit will be blocked until they are resolved.

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
[#158757927](https://www.pivotaltracker.com/story/show/158757927)

#### Screenshots (if appropriate)
- N/A

#### Questions:
- N/A
